--- a/src/Element/Select2.php
+++ b/src/Element/Select2.php
@@ -213,7 +213,7 @@
       'tags' => (bool) $element['#autocreate'],
       'theme' => $select2_theme_exists ? $current_theme : 'default',
       'maximumSelectionLength' => $multiple ? $element['#cardinality'] : 0,
-      'tokenSeparators' => $element['#autocreate'] ? [','] : [],
+      'tokenSeparators' => $element['#autocreate'] ? ['|'] : [],
       'selectOnClose' => $element['#autocomplete'],
       'width' => '100%',
     ];
